lib/LaTeXML/Core/Document.pm

```diff
@@ -1376,8 +1376,14 @@ sub makeError {
 # [xml:id and namespaced attributes are always allowed]
 sub setAttribute {
   my ($self, $node, $key, $value) = @_;
-  return                       if (ref $value) && ((!blessed($value)) || !$value->can('toAttribute'));
-  $value = $value->toAttribute if ref $value;
+  return if (ref $value) && ((!blessed($value)) || !$value->can('toAttribute'));
```

Comment:
Would it be useful to emit a `Warning` if this unusual case is encountered, where `$value` was e.g. a perl array ref or hash ref? Could be some rare mistake that gets caught quickly with a warning message.